File: Selenium/Project/task4.java
package project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class task4 {

    WebDriver driver;
    // Verify the title of the second most popular course
    // Goal: Read the title of the second most popular course on the website and verify the text

    @BeforeClass
    public void beforeMethod() {
        //Set up the firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        driver = new FirefoxDriver();

        // Open a browser.
        // Navigate to ‘https://alchemy.hguy.co/lms’
        driver.get("https://alchemy.hguy.co/lms");
    }

    @Test
    public void secondMostPopularCourse() {
        // Get the title of the second most popular course
        WebElement title = driver.findElement(By.xpath("//*[@id='post-71']/div[2]/h3"));
        String titleOfSecondCourse;
        titleOfSecondCourse = title.getText();
        System.out.println("Title of the second most popular course is: " + titleOfSecondCourse);

        // Make sure it matches “Email Marketing Strategies” exactly
        Assert.assertEquals(titleOfSecondCourse, "Email Marketing Strategies");
    }

    @AfterClass
    public void afterMethod() {
        // If it matches, close the browser
        driver.quit();
    }
}

File: Selenium/Project/task4.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Verify the title of the second most popular course
# Goal: Read the title of the second most popular course on the website and verify the text


# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to ‘https://alchemy.hguy.co/lms’.
    driver.get("https://alchemy.hguy.co/lms/")

    # Get the title of the second most popular course
    Title = driver.find_element(By.XPATH,"//*[@id='post-71']/div[2]/h3")
    TitleOfSecondPopularCourse = Title.text

    print("Title of second most popular course is: ", TitleOfSecondPopularCourse)

    # Make sure it matches "Email Marketing Strategies" exactly
    assert TitleOfSecondPopularCourse == "Email Marketing Strategies"

    # If it matches, close the browser
    driver.quit